### DISCUSSION TITLE
Emulator for displays on desktop?

### DISCUSSION THREAD

**Discussion by oschettler** [tinygo-org/tinygo#4112] [Q&A]
Is it possible to target my desktop computer and even have emulated buttons and display (320x240 pixels), then later port the code to a microcontroller, eg. to a Wio Terminal?

**Comment by sago35** [comment]
I have created the following. I haven't used it in a while, but I think it should work. @oschettler
https://github.com/sago35/tinydisplay
![tinydisplay_pyportal_boing](https://github.com/tinygo-org/tinygo/assets/9251039/9fdf22ca-4486-4e0f-99ea-c8bdba5734b8)

**Reply by oschettler** [reply]
Sorry, I am an utter beginner with both Go and TinyGo. 

When trying any of the examples, I run into 

````
olav@hp2:~/Dokumente/tinydisplay/examples/basic$ tinygo run main.go 
# net/rpc
/usr/local/go/src/net/rpc/server.go:291:14: method.IsExported undefined (type reflect.Method has no field or method IsExported)
````

This very same message is also mentioned in #3992.

/usr/local/go contains an install of go1.21.6 (output of "go version")
My "tinygo version" says: "tinygo version 0.30.0 linux/amd64 (using go version go1.21.6 and LLVM version 16.0.1)"

**Reply by oschettler** [reply]
Ah, but just following your README, I could flash my Wio Terminal:

![IMG_8392](https://github.com/tinygo-org/tinygo/assets/124634/edfbb241-d9e1-4a79-80f7-5c5b1844ec54)

So, never mind the reflect problem. Thank you again.

**Comment by oschettler** [comment]
@sago35 awesome, thank you, I'll have a look. 

I have also seen [Ayke's post on Playground in VS Code](https://aykevl.nl/2022/05/tinygo-preview/), but I have not seen any examples for boards with displays.
